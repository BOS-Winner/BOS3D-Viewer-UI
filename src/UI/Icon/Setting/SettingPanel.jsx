import React from "react";
import CameraSetting from "./CameraSetting";
import ModelSetting from "./ModelSetting";
import DisplayAndEffect from "./DisplayAndEffect";
import ToolbarSetting from "./ToolSetting";
import BottomBar from "./BottomBar";
import style from "./style.less";
import { AntdIcon } from '../../utils/utils';

class SettingPanel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      settingItem: 'model', // model模型设置，camera相机设置
    };
  }

  switchSetItem(name, e) {
    e.preventDefault();
    e.stopPropagation();
    this.setState({
      settingItem: name,
    });
  }

  render() {
    const { settingItem } = this.state;
    const { isMobile } = this.props;

    return (
      <div className={style.settingPanel}>
        <div
          className={style.tabsWrap}
        >
          <div
            className={style.tabs}
          >
            <div
              role="presentation"
              className={`${style.tab} ${settingItem === 'model' ? style.active : ''}`}
              onClick={e => { this.switchSetItem('model', e) }}
            >
              <AntdIcon type="iconmodelsettings" />
              {' '}
              模型设置
            </div>
            <div
              role="presentation"
              className={`${style.tab} ${settingItem === 'camera' ? style.active : ''}`}
              onClick={e => { this.switchSetItem('camera', e) }}
            >
              <AntdIcon type="iconcamerasettings" />
              {' '}
              相机设置
            </div>
            <div
              role="presentation"
              className={`${style.tab} ${settingItem === 'display' ? style.active : ''}`}
              onClick={e => { this.switchSetItem('display', e) }}
            >
              <AntdIcon type="icondisplayandeffect" />
              {' '}
              显示与效果
            </div>
            <div
              role="presentation"
              className={`${style.tab} ${settingItem === 'toolbar' ? style.active : ''}`}
              onClick={e => { this.switchSetItem('toolbar', e) }}
            >
              <AntdIcon type="icontoolconfiguration" />
              {' '}
              工具栏配置
            </div>
          </div>
        </div>
        <div className={style.tabPanel}>
          <div
            style={{
              display: this.state.settingItem === 'model' ? 'block' : 'none'
            }}
          >
            <ModelSetting isMobile={isMobile} />
          </div>
          <div
            style={{
              display: this.state.settingItem === 'camera' ? 'block' : 'none'
            }}
          >
            <CameraSetting isMobile={isMobile} />
          </div>
          <div
            style={{
              display: this.state.settingItem === 'display' ? 'block' : 'none'
            }}
          >
            <DisplayAndEffect isShown={this.state.settingItem === 'display'} isMobile={isMobile} />
          </div>
          <div
            style={{
              display: this.state.settingItem === 'toolbar' ? 'block' : 'none'
            }}
          >
            <ToolbarSetting />
          </div>
          {this.state.settingItem !== 'model' && <BottomBar isMobile={isMobile} type={this.state.settingItem} />}
        </div>

      </div>
    );
  }
}

export default SettingPanel;
