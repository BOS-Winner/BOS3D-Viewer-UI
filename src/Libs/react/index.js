import React from "react";
import useUpdateEffect from "./useUpdateEffect";

React.useUpdateEffect = useUpdateEffect;

export default React;
// export { default as useUpdateEffect } from "./useUpdateEffect";
