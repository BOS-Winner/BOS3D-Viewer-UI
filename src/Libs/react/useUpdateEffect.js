import { useEffect, useRef } from "react";

export default function (effect, deps) {
  const isFirstRender = useRef(true);
  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else {
      effect();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, deps);
}
