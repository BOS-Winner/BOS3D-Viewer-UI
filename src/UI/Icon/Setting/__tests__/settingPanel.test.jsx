import React from "react";
import { render, act } from "@testing-library/react";
import SettingPanel from "../SettingPanel.jsx";

jest.mock('../ModelSetting', () => () => <div title="ModelSetting" />);
jest.mock('../CameraSetting', () => () => <div title="CameraSetting" />);
jest.mock('../DisplayAndEffect', () => () => <div title="DisplayAndEffect" />);
jest.mock('../ToolSetting', () => () => <div title="ToolSetting" />);
jest.mock('../BottomBar', () => () => <div title="BottomBar" />);

let spContainer;
let spGetByText;

describe("setting panel", () => {
  beforeEach(() => {
    const { container, getByText } = render(<SettingPanel />);
    spContainer = container;
    spGetByText = getByText;
  });

  it("should switch panel", () => {
    act(() => {
      spGetByText('相机设置').click();
    });
    expect(spContainer).toMatchSnapshot();
    act(() => {
      spGetByText('显示与效果').click();
    });
    expect(spContainer).toMatchSnapshot();
    act(() => {
      spGetByText('工具栏配置').click();
    });
    expect(spContainer).toMatchSnapshot();
    act(() => {
      spGetByText('模型设置').click();
    });
    expect(spContainer).toMatchSnapshot();
  });
});
