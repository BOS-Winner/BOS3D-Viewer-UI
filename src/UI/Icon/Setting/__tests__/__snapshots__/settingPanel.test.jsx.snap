// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setting panel should switch panel 1`] = `
<div>
  <div
    class="settingPanel"
  >
    <div
      class="switch"
      role="button"
      tabindex="0"
    >
      <div
        class="switchItem"
        data-name="model"
      >
        模型设置
      </div>
      <div
        class="switchSelect"
        data-name="camera"
      >
        相机设置
      </div>
      <div
        class="switchItem"
        data-name="display"
      >
        显示与效果
      </div>
      <div
        class="switchItem"
        data-name="toolbar"
      >
        工具栏配置
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ModelSetting"
      />
    </div>
    <div
      style="display: block;"
    >
      <div
        title="CameraSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="DisplayAndEffect"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ToolSetting"
      />
    </div>
    <div
      title="BottomBar"
    />
  </div>
</div>
`;

exports[`setting panel should switch panel 2`] = `
<div>
  <div
    class="settingPanel"
  >
    <div
      class="switch"
      role="button"
      tabindex="0"
    >
      <div
        class="switchItem"
        data-name="model"
      >
        模型设置
      </div>
      <div
        class="switchItem"
        data-name="camera"
      >
        相机设置
      </div>
      <div
        class="switchSelect"
        data-name="display"
      >
        显示与效果
      </div>
      <div
        class="switchItem"
        data-name="toolbar"
      >
        工具栏配置
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ModelSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="CameraSetting"
      />
    </div>
    <div
      style="display: block;"
    >
      <div
        title="DisplayAndEffect"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ToolSetting"
      />
    </div>
    <div
      title="BottomBar"
    />
  </div>
</div>
`;

exports[`setting panel should switch panel 3`] = `
<div>
  <div
    class="settingPanel"
  >
    <div
      class="switch"
      role="button"
      tabindex="0"
    >
      <div
        class="switchItem"
        data-name="model"
      >
        模型设置
      </div>
      <div
        class="switchItem"
        data-name="camera"
      >
        相机设置
      </div>
      <div
        class="switchItem"
        data-name="display"
      >
        显示与效果
      </div>
      <div
        class="switchSelect"
        data-name="toolbar"
      >
        工具栏配置
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ModelSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="CameraSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="DisplayAndEffect"
      />
    </div>
    <div
      style="display: block;"
    >
      <div
        title="ToolSetting"
      />
    </div>
    <div
      title="BottomBar"
    />
  </div>
</div>
`;

exports[`setting panel should switch panel 4`] = `
<div>
  <div
    class="settingPanel"
  >
    <div
      class="switch"
      role="button"
      tabindex="0"
    >
      <div
        class="switchSelect"
        data-name="model"
      >
        模型设置
      </div>
      <div
        class="switchItem"
        data-name="camera"
      >
        相机设置
      </div>
      <div
        class="switchItem"
        data-name="display"
      >
        显示与效果
      </div>
      <div
        class="switchItem"
        data-name="toolbar"
      >
        工具栏配置
      </div>
    </div>
    <div
      style="display: block;"
    >
      <div
        title="ModelSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="CameraSetting"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="DisplayAndEffect"
      />
    </div>
    <div
      style="display: none;"
    >
      <div
        title="ToolSetting"
      />
    </div>
  </div>
</div>
`;
